version: '3.7'

services:

  app:
    build:
      context: .
      dockerfile: .docker/Dockerfile
      args:
        APP_ENV: ${APP_ENV}
    env_file:
      - .env
    restart: on-failure
    working_dir: /app
    entrypoint: ["./.docker/init.sh"]
    ports:
      - '8888:80'
    volumes:
      - .:/app
      - ./.docker/nginx/nginx.conf:/etc/nginx/nginx.conf
      - ./.docker/nginx/vhost.conf:/etc/nginx/sites-enabled/default
    depends_on:
      - mysql

  mysql:
    image: mysql:5.7
    command: --default-authentication-plugin=mysql_native_password --sql_mode=STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION
    restart: always
    tty: true
    ports:
      - "23306:3306"
    volumes:
      - ./.docker/mysql:/var/lib/mysql
      #- ./.docker/shared:/home/shared
    environment:
      MYSQL_USER: ${DATABASE_USER}
      MYSQL_DATABASE: ${DATABASE_DATABASE}
      MYSQL_PASSWORD: ${DATABASE_PASSWORD}
      MYSQL_ROOT_PASSWORD: ${DATABASE_ROOT_PASSWORD}
